### DISCUSSION TITLE
✨ core function to get a directory or file as tarball

### DISCUSSION THREAD

**Discussion by gjung56** [dagger/dagger#12133] [Ideas]
### What are you trying to do?

It would be nice to have a core function to get a tarball from a directory or a file

Methods api examples : 
Directory.AsTarball()
File.AsTarball() 

And eventually, the reverse functions, to get a directory from a tarball.

### Why is this important to you?

We use a lot tarballs in our ci/cd pipelines and that make a lot of the same boilerplate to get what we need.

### How are you currently working around this?

I use a basic alpine container, mount the source directory and exec a tar command then export the resulted file.

**Comment by jedevc** [comment]
We should also support unpacking files/directories as part of this.

**Comment by realoriginal** [comment]
I'd love for this. I'm running into the issue described here https://github.com/dagger/dagger/issues/6355 and this would be an excellent workaround.
